utils.bal

```diff
@@ -0,0 +1,723 @@
+    }
+}
+
+# Upload new file.
+# 
+# + httpClient - The HTTP Client
+# + filePath - Path to the file object to be uploaded
+# + optional - 'UpdateFileMetadataOptional' used to add query parameters to the request
+# + fileMetadata - 'File' Metadata is send to in the payload 
+# + return - If successful, returns `File`. Else returns `error`
+function uploadFile(http:Client httpClient, string filePath, UpdateFileMetadataOptional? optional = (), 
+                                File? fileMetadata = ()) returns @tainted File|error {    
+    string path = prepareUrl([UPLOAD, DRIVE_PATH, FILES]);  
+    json response = check uploadFiles(httpClient, path, filePath);  
+    //update metadata
+    json|error respId = response.id;
+    string fileId = EMPTY_STRING;
+    if (respId is json) {
+        fileId = respId.toString();
+    }
+    string newFileUrl = prepareUrlWithUpdateOptional(fileId, optional);
+    json payload = check fileMetadata.cloneWithType(json);
+    json changeResponse = check updateRequestWithPayload(httpClient, newFileUrl, payload);
+    File|error file = changeResponse.cloneWithType(File);
+    if (file is File) {
+        return file;
+    } else {
+        return error(ERR_FILE_RESPONSE, file);
+    }
+}
+
+# Retrieve files.
+# 
+# + httpClient - The HTTP Client
+# + optional - 'ListFilesOptional' used to add query parameters to the request
+# + return - If successful, returns stream of files `stream<File>`. Else returns `error`
+function getFiles(http:Client httpClient, ListFilesOptional? optional = ()) returns @tainted stream<File>|error {
+    string path = prepareUrlwithFileListOptional(optional);
+    json response = check sendRequest(httpClient, path);
+    File[] files = [];
+    FilesResponse|error res = response.cloneWithType(FilesResponse);
+    if (res is FilesResponse) {
+        int i = files.length();
+        foreach File item in res.files {
+            files[i] = item;
+            i = i + 1;
+        }        
+        stream<File> filesStream = (<@untainted>files).toStream();
+        return filesStream;
+    } else {
+        return error(ERR_FILE_RESPONSE, res);
+    }
+}
+
+# Upload new file using a Byte array.
+# 
+# + httpClient - The HTTP Client
+# + byteArray - Byte array that represents the file object
+# + optional - 'UpdateFileMetadataOptional' used to add query parameters to the request
+# + fileMetadata - 'File' Metadata is send to in the payload 
+# + return - If successful, returns `File`. Else returns `error`
+function uploadFileUsingByteArray(http:Client httpClient, byte[] byteArray, UpdateFileMetadataOptional? optional = (), 
+                                  File? fileMetadata = ()) returns @tainted File|error {    
+    string path = prepareUrl([UPLOAD, DRIVE_PATH, FILES]);
+    log:print(path.toString());
+    json response = check uploadFileWithByteArray(httpClient, path, byteArray);
+    //update metadata
+    json|error respId = response.id;
```

Comment:
Shall we use a better name than respId?

Reply:
Changed to 'responseId' 
Thanks @LakshanSS